### PR TITLE
new E decomposition without svd

### PR DESCRIPTION
**Pull request by weitong8591**
#### Changes
As SVD used in E decomposition is not stable for training, we would like to intergrate another methodthat recovers the two solutions for the baseline and the orientation from a given essential matrix using elementary matrix operations. This work was done by [Berthold K.P. Horn in 1990](https://people.csail.mit.edu/bkph/articles/Essential.pdf), and used in Generalized Differentiable RANSAC to build the pose erorr loss. 

Fixes # (issue)

#### Type of change

- [x] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📝 This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Comment by weitong8591** [comment]
Hi! any suggestions about what to return if there is no solution?  It can't pass the tests. [link](https://github.com/weitong8591/kornia/blob/e8729cc77c2b9c3569c5d9d1b71fffbfd2be1d6a/kornia/geometry/epipolar/numeric.py#L47)

**Comment by johnnv1** [comment]
> Hi! any suggestions about what to return if there is no solution? It can't pass the tests. [link](https://github.com/weitong8591/kornia/blob/e8729cc77c2b9c3569c5d9d1b71fffbfd2be1d6a/kornia/geometry/epipolar/numeric.py#L47)

since we are randomly generating the input, we cannot determine when it'll be able to solve or not... i may say the best case here is to mark the test as xfail -- https://docs.pytest.org/en/stable/how-to/skipping.html#raises-parameter

File: kornia/geometry/epipolar/essential.py [hunk 1]
```diff
@@ -327,6 +328,68 @@ def decompose_essential_matrix(E_mat: torch.Tensor) -> Tuple[torch.Tensor, torch
     return (R1, R2, T)
 
 
+def decompose_essential_matrix_no_svd(E_mat: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+    r"""
+      recover rotation and translation from essential matrices without SVD
+      reference: Horn, Berthold KP. Recovering baseline and orientation from essential matrix[J].
+      J. Opt. Soc. Am, 1990, 110.
```
Comment (edgarriba) [inline kornia/geometry/epipolar/essential.py:335]:
```suggestion
      J. Opt. Soc. Am, 1990, 110.
      
```

File: kornia/geometry/epipolar/essential.py [hunk 2]
```diff
@@ -327,6 +328,68 @@ def decompose_essential_matrix(E_mat: torch.Tensor) -> Tuple[torch.Tensor, torch
     return (R1, R2, T)
 
 
+def decompose_essential_matrix_no_svd(E_mat: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+    r"""
```
Comment (edgarriba) [inline kornia/geometry/epipolar/essential.py:332]:
```suggestion
    r"""Decompose the essential matrix to rotation and translation.
    
```

**Review by edgarriba — CHANGES_REQUESTED** [review]

File: kornia/geometry/epipolar/essential.py [hunk 3]
```diff
@@ -327,6 +328,68 @@ def decompose_essential_matrix(E_mat: torch.Tensor) -> Tuple[torch.Tensor, torch
...
+
+    Returns:
+       A tuple containing the first and second possible rotation matrices and the translation vector.
+       The shape of the tensors with be same input :math:`[(*, 3, 3), (*, 3, 3), (*, 3, 1)]`.
+
+    """
+    if not (len(E_mat.shape) >= 2 and E_mat.shape[-2:]):
+        raise AssertionError(E_mat.shape)
+
+    B = E_mat.shape[0]
```
[the remark this answers is not in the document]
Comment (weitong8591) [reply kornia/geometry/epipolar/essential.py:347]: Sounds good, i will change it.

**Comment by weitong8591** [comment]
> > Hi! any suggestions about what to return if there is no solution? It can't pass the tests. [link](https://github.com/weitong8591/kornia/blob/e8729cc77c2b9c3569c5d9d1b71fffbfd2be1d6a/kornia/geometry/epipolar/numeric.py#L47)
> 
> since we are randomly generating the input, we cannot determine when it'll be able to solve or not... i may say the best case here is to mark the test as xfail -- https://docs.pytest.org/en/stable/how-to/skipping.html#raises-parameter

Thanks! let me try it

File: kornia/geometry/epipolar/essential.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (weitong8591) [reply kornia/geometry/epipolar/essential.py:350]: What about making it like, 
`elif len(E_mat.shape) != 3: 
        E_mat = E_mat.view(-1, 3, 3)`

**Review by edgarriba — APPROVED** [review]

**Pull request merged**